Node/src/botbuilder-teams.d.ts

```diff
@@ -772,16 +772,175 @@ export interface IComposeExtensionResponse {
+
+  /** Represents details of the converesation. */
+  conversation: IMessageActionsPayloadConversation;
+}
+
+/** Represents a user entity. */
+export interface IMessageActionsPayloadUser {
+  /** The identity type of the user. */
+  userIdentityType: 'aadUser' | 'onPremiseAadUser' | 'anonymousGuest' | 'federatedUser';
+
+  /** The id of the user. */
+  id: string;
+
+  /** The plaintext display name of the user. */
+  displayName: string;
+}
+
+/** Represents an application entity. */
+export interface IMessageActionsPayloadApp {
+  /** The type of application. */
+  applicationIdentityType:  'aadApplication' | 'bot' | 'tenantBot' | 'office365Connector' | 'webhook';
+
+  /** The id of the application. */
+  id: string;
+
+  /** The plaintext display name of the application. */
+  displayName: string;
+}
+
+/** Represents a team or channel entity. */
+export interface IMessageActionsPayloadConversation {
+  /** The type of conversation, whether a team or channel. */
+  conversationIdentityType: 'team' | 'channel';
+
+  /** The id of the team or channel. */
+  id: string;
+  
+  /** The plaintext display name of the team or channel entity. */
+  displayName: string;
+}
+
+/** Represents the entity that was mentioned in the message. */
+export interface IMessageActionsPayloadMention {
+  /** The id of the mentioned entity. */
+  id: number;
+
+  /** The plaintext display name of the mentioned entity. */
+  mentionText: string;
+
+  /** Provides more details on the mentioned entity.  */
+  mentioned: IMessageActionsPayloadFrom;
+}
+
+/** Represents the reaction of a user to a message. */
+export interface IMessageActionsPayloadReaction {
+  /** The type of reaction given to the message. */
+  reactionType: 'like' | 'heart' | 'laugh' | 'surprised' | 'sad' | 'angry';
+
+  /** Timestamp of when the user reacted to the message. */
+  createdDateTime: string;
+
+  /** The user with which the reaction is associated. */
+  user: IMessageActionsPayloadFrom;
+}
+
+/** Represents the attachment in a message. */
+export interface IMessageActionsPayloadAttachment {
+  /** The id of the attachment. */
+  id: string;
+  
+  /** The type of the attachment. */
+  contentType: string;
+
+  /** The url of the attachment, in case of a external link. */
+  contentUrl?: string;
+
+  /** The content of the attachment, in case of a code snippet, email, or file. */
+  content?: any;
+
+  /** The plaintext display name of the attachment. */
+  name?: string;
+
+  /** The url of a thumbnail image that might be embedded in the attachment, in case of a card.  */
+  thumbnailUrl?: string;
+}
+
 /** Represents the value of the invoke activity of compose extension action command request */
 export interface IComposeExtensionActionCommandRequest extends ITaskModuleInvokeRequest {
   /** The id of the command. */
   commandId?: string;
 
+  /** The context from which the command originates. */
+  commandContext: 'message' | 'compose' | 'commandbox';
```

Comment:
optional?

Reply:
Yeah, just double-checked the FE code, it should be.